File: gaia-sdk-python/gaia_sdk/graphql/response/type/CreatedSkillImpulse.py

from gaia_sdk.graphql.response.type.Skill import Skill

from dataclasses import dataclass
from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool
from gaia_sdk.graphql.request.enumeration.RuntimeState import RuntimeState
from gaia_sdk.graphql.request.enumeration.SkillState import SkillState
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField

@dataclass
class CreatedSkillImpulse:
    """
    Impulse which indicates the result of a create Skill impulse
    """
    dictionary: dict
    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    the Skill instance
    """
    @property
    def data(self) -> Skill:
        return Skill(self.dictionary.get("data"))

File: gaia-sdk-javascript/src/graphql/response/type/CreatedSkillImpulse.ts

import {Skill} from "./Skill";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../../request/enumeration/RuntimeState";
import {SkillState} from "../../request/enumeration/SkillState";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a create Skill impulse
*/
export interface CreatedSkillImpulse {
    id?:Uuid, 
    /**
    * the Skill instance
    */
    data?:Skill
}